common/scanner.h

```diff
@@ -27,27 +32,37 @@ static bool scan_pi_target(TSLexer *lexer) {
     }
 
     if (advanced_once) {
-        while (isalnum(lexer->lookahead) || lexer->lookahead == '_' ||
-               lexer->lookahead == ':' || lexer->lookahead == '.' ||
-               lexer->lookahead == L'·' || lexer->lookahead == '-') {
-            if (lexer->lookahead == 'x' || lexer->lookahead == 'X') {
-                lexer->mark_end(lexer);
-                advance(lexer);
-                if (lexer->lookahead == 'm' || lexer->lookahead == 'M') {
-                    advance(lexer);
-                    if (lexer->lookahead == 'l' || lexer->lookahead == 'L') {
-                        advance(lexer);
-                        return false;
-                    }
-                }
-            }
-
+        while (is_valid_pi_char(lexer->lookahead)) {
             if (found_x_first &&
                 (lexer->lookahead == 'm' || lexer->lookahead == 'M')) {
                 advance(lexer);
                 if (lexer->lookahead == 'l' || lexer->lookahead == 'L') {
                     advance(lexer);
-                    return false;
+                    if (is_valid_pi_char(lexer->lookahead)) {
+                        found_x_first = false;
+                        bool last_char_hyphen = lexer->lookahead == '-';
+                        advance(lexer);
+                        if (last_char_hyphen) {
+                            // scan for stylesheet/model and disallow that
```

Comment:
It's not actually an error to allow those.

Reply:
it was overriding them in the tests

Reply:
Does increasing the precedence not work?

Reply:
throwing the string literal as an external without processing it and checking if it's valid could work

Reply:
There has to be a better solution...

Reply:
There really isn't :grimacing:, either xml-model in the scanner will override the literal string unconditionally or we disallow it at all or we check if the string is valid and then disallow it if it is valid